build-docker-image.sh

```diff
@@ -62,3 +72,8 @@ if [[ "$NAME" != jenkins-* || "$NAME" != latest ]]; then
 fi
 rm -f ___version___
 docker push "$DOCKER_REGISTRY/$NAME:$LABEL"
+if [ "$push_external" -eq 1 ]; then
+    trap "docker rmi $DOCKER_EXTERNAL_REGISTRY/$NAME:$LABEL" EXIT
```

Comment:
Is this just being over cautious, or do the supported external registries not allow overwrite ?

Reply:
This is removing the image from the local machine, not the remote registry, to avoid filling up the build machine with every image ever built.

Reply:
Of course - was looking at this too late at night and assumed it was undoing the push...